### PR TITLE
Add Companion WTR001 Temperature Sensor

### PR DESCRIPTION
**Pull request by klohner**
Add support for Companion WTR001 Temperature Sensor.  Decodes known .cu8 sample files.

### CODE REVIEW

File: src/devices/companion_wtr001.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Published range of device is -29.9C to 69.9C
+*/
+
+#include "decoder.h"
+
+#define MYDEVICE_BITLEN      14
+#define MYDEVICE_MINREPEATS  3
+
+static int companion_wtr001_decoder(r_device *decoder, bitbuffer_t *bitbuffer)
```
Comment (zuckschwerdt) [inline src/devices/companion_wtr001.c]: Preferably this should be …_decode

File: src/devices/companion_wtr001.c [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Whole part of Temperature is too high (%d - 41 is greater than 70)\n", temp_whole_raw);
+        }
+        return 0;
+    }
+   
+    temperature = (temp_whole_raw + (temp_tenth_raw * 0.1f)) - 41.0f;
+
+    data = data_make(
+            "model", "", DATA_STRING, _X("WTR001","Companion WTR001 Temperature Sensor"),
```
Comment (zuckschwerdt) [inline src/devices/companion_wtr001.c]: No need for a deprecated "old model" name and _X macro. "Companion-WTR001" sounds right.

**Comment by zuckschwerdt** [comment]
Looks great! Will be interesting to see if something breaks with the missing `id` ;)

File: src/devices/companion_wtr001.c [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    data_t *data;
+    int r; // a row index
+    uint8_t b[2];
+    float temperature;
+
+    // short pulse with long gap is 0, long pulse with short gap is 1.
+    // This is inverse of default OOK_PULSE_PWM behavior
+    bitbuffer_invert(bitbuffer);
+
```
Comment (merbanan) [inline src/devices/companion_wtr001.c]: Do bitbuffer_invert() later.

File: src/devices/companion_wtr001.c [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    if (r < 0 || bitbuffer->bits_per_row[r] != 14) {
+        return 0;
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, r, 0, b, 14);
+
+    // Make sure fixed bit in nibble 5 is not set
+    if ((b[0] & 0x04) == 0x04) {
+        if (decoder->verbose > 1) {
+            // fprintf(stderr, "companion_wtr001: Fixed Bit set (and it shouldn't be)\n");
```
Comment (merbanan) [inline src/devices/companion_wtr001.c]: remove verbose check or enable fprintf

File: src/devices/companion_wtr001.c [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "companion_wtr001: Temperature Degree Tenth too high (%d - 10 is greater than 9\n", temp_tenth_raw);
+        }
+        return 0;
+    }
+
+    temp_tenth_raw -= 0x0a;
+    
+    uint8_t temp_whole_raw = reverse8(b[1]&0xf0) | reverse8(b[0]&0x03)>>2 | (b[1]&0x08)<<3;
+
```
Comment (merbanan) [inline src/devices/companion_wtr001.c]: Maybe add bin2dec(bits 12,7,6,11,10,9,8) as a comment above this line.

**Comment by merbanan** [comment]
Really nice and clean code. And this was a fun puzzle to crack.

File: src/devices/companion_wtr001.c (hunk 4 above)
Comment (klohner) [reply src/devices/companion_wtr001.c]: Enabled the fprintf.

**Comment by klohner** [comment]
Thanks for all the comments.  I've addressed them all.  This was fun!

**Comment by zuckschwerdt** [comment]
@merbanan somehow this wasn't squash merged and also didn't record the issue number (same with last merge). I'll fix this with a history rewrite now.
Heads up, everybody already on current master needs: `git reset --hard HEAD~8 ; git pull`

**Pull request merged**